airflow/providers/apache/livy/hooks/livy.py

```diff
@@ -575,10 +575,11 @@ def _generate_base_url(self, conn: Connection) -> str:
         if conn.host and "://" in conn.host:
             base_url: str = conn.host
         else:
-            # schema defaults to HTTP
-            schema = conn.schema if conn.schema else "http"
+            scheme = conn.conn_type
```

Comment:
You can not set connection type value in the UI, the only possible way is a select one from drop down menu. 
There is no `https` connection type registered into the core/community providers.

For example the Apache Livy Connection will use `livy`, HTTP will use `http`, for Airfbyte will use `airbyte`, dbt Cloud will use `dbt_cloud` and the same in others hooks which based on HttpHook

Reply:
I see. In that case, do you think it would make sense to change connections as follows:
- register 'https' as a separate connection type which inherits from HTTP with the only difference in conn_type so that it becomes visible in UI
- for other hooks based on HttpHook, make them use connections of type 'http' and 'https' instead of custom types. I understand that these integrations need custom hook implementation, but what is the point of custom connection types?

Reply:
Because it could handle connection by the different way.
I'm still confuse why it required to have `https` connection type, what is a benefit?

Reply:
It could use connection in a different way, but this difference doesn't change the connection type. It so happens that my org also uses Airbyte hook - I just checked and we use `http` connection type (not `airbyte`) for AirbyteHook and it works perfectly fine. It was accidental but it actually makes sense to me - just because AirbyteHook has some custom logic on top of HTTP connection doesn't make this connection anything else than HTTP because that handling is a higher level of abstraction than connection.

I can see a couple of benefits now: 
1) The tactical one is that this change fixes a bug https://github.com/apache/airflow/issues/10913 where you can't correctly set path for HTTP connection if creating connection from Airflow URI because path from URI is used as protocol in the hook. In the process of fixing it, you can
2) More importantly - UX:
a) easier connection creation for new users - even though Airflow URI in principle is not a valid HTTP(S) URI, users can expect if to be interpreted as such for HTTP connections - nothing in the URI syntax in docs suggests that connection to https://example.com should be defined http://example.com/https so it's completely counter-intuitive
b) wouldn't it be easier for users to have fewer connection types to choose in the drop-down list? In practice all they need to choose for those cases is HTTP/HTTPS

Reply:
>  I just checked and we use http connection type (not airbyte) 

The main point here "you use", don't forget about other users or some one who create Connection from the UI or use JSON connection. Some of the connections also use additional fields, which not provided by HTTP connection, in additional HTTP connection provide additional configuration thought Extra field

> easier connection creation for new users

URI not the only one way to define connection, and to be honest not he easiest way to create a connection, due to requirements of decode it

![image](https://github.com/apache/airflow/assets/3998685/d884f0d4-907d-404f-94f6-a431b3f9be94)

> even though Airflow URI in principle is not a valid HTTP(S) URI, users can expect if to be interpreted as such for HTTP connections

And we return to initial comment https://github.com/apache/airflow/pull/35712#pullrequestreview-1737927952

This mentioned into the documentation. and maybe better to move it into the [HTTP operator](https://airflow.apache.org/docs/apache-airflow-providers-http/stable/operators.html#httpoperator), maybe better move or duplicate it into the Connection.

> wouldn't it be easier for users to have fewer connection types to choose in the drop-down list? In practice all they need to choose for those cases is HTTP/HTTPS

This action is required to create additional Hook for handle this "feature".

Reply:
> Some of the connections also use additional fields
Http-based hooks don't use additional fields, some of them hide or re-label fields but it's cosmetic - no functionality is missing without this.

>  URI not the only one way to define connection, and to be honest not he easiest way to create a connection
This is subjective. JSON format is newer but both have its uses. People who specify connection with env variables will want a one-liner. And URI format is still supported by Airflow. Also, JSON format has problematic names. Suppose you want to set up connection to `https://example.com/endpoint`, then configuration would be like this:
```
"example_https": {
  "conn_type": "http",
  "host": "https://example.com/endpoint"
}
```
but clearly `https://example.com/endpoint` is not a 'host', it's a URL so you need to be experienced with JSON format to do it correctly. After the change you could write it:
```
"example_https": {
  "conn_type": "https",
  "host": "example.com"
  "schema": "endpoint"
}
```
(or in the URI format).

> This action is required to create additional Hook for handle this "feature".
Can you elaborate what do you mean?

Reply:
If Connection URI is not supposed to follow URI generic syntax standard like RFC 3986, then maybe it should be called "connection string" or sth like this and not "URI".

Reply:
Anyway, I see that this change is bigger in scope than I thought so it probably needs a devlist vote or sth like this.